ta4j-core/src/test/java/org/ta4j/core/TestUtils.java

```diff
@@ -22,96 +22,214 @@ this software and associated documentation files (the "Software"), to deal in
  */
 package org.ta4j.core;
 
-import org.ta4j.core.num.Num;
+import java.util.function.Function;
 
-import java.math.BigDecimal;
-import java.util.List;
+import org.ta4j.core.num.BigDecimalNum;
+import org.ta4j.core.num.DoubleNum;
+import org.ta4j.core.num.Num;
 
 import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotEquals;
 
 /**
- * Utility class for {@code Decimal} tests.
+ * Utility class for {@code Num} tests.
  */
 public class TestUtils {
 
+    // digits of precision for high precision operations
+    public static final int HIGH_PRECISION = 64;
+    // override the default BigDecimalNum precision for high precision conversions
+    private static final Function<String, Num> highPrecisionNumFunc = (val -> BigDecimalNum.valueOf(val, HIGH_PRECISION));
     /** Offset for double equality checking */
-    public static final double GENERAL_OFFSET = 0.0001;
+    // deprecated so that JUnit Assert calls direct from unit tests will show warnings
+    // TODO: modify unit tests to not call JUnit Assert but to call TestUtils Assert
+    @Deprecated
+    public static final double GENERAL_OFFSET = DoubleNum.valueOf(DoubleNum.EPS).doubleValue();
+    private static Num HIGH_PRECISION_OFFSET = highPrecisionNumFunc.apply("0.0001");
+
+    public static void setPrecision(String precision) {
+        HIGH_PRECISION_OFFSET = highPrecisionNumFunc.apply(precision);
```

Comment:
This was lazy.  Should not be a static method to update a static field.  Should probably just add signatures for methods that accept an explicit precision parameter.